### ISSUE TITLE
[Brave Leo]: Deprecate Claude Instant model

### ISSUE THREAD

**Issue by LorenzoMinto** [brave/brave-browser#37988]
Should migrate Claude Instant defaults to Claude Haiku

## Test Plan:
- Not premium user
- Using a version before this commit, set Claude Instant to be the default
- Verify every new conversation uses Claude Instant
- Quit the browser and open again using this commit, with the same profile
- Verify that every new conversation uses Claude Haiku
- Repeat for premium user except every new conversation should now use Claude Sonnet

**Comment by petemill** [comment]
I can take this to remove claude instant and migrate the prefs

**Comment by kjozwiak** [comment]
The above requires `1.67.102` or higher for `1.67.x` verification 👍

**Comment by kjozwiak** [comment]
@hffvld @Uni-verse we'll need to wait till https://github.com/brave/brave-core/pull/23911 is merged into `master` and uplifted into `1.67.x` before we can run through the above on `iOS`.

**Comment by Uni-verse** [comment]
Verified on `Samsung Galaxy S21` using version: 

```
Brave	1.67.104 Chromium: 126.0.6478.26 (Official Build) beta (64-bit) 
Revision	9a4d1cb2f47a60953416cd3f39f0520e4dd4cf0e
OS	Android 13; Build/TP1A.220624.014; 33; REL
```

**Case: New Users**

- Ensured Claude instant is not available in the Language models section in the settings menu for new users. 
- Ensured Claude Instant is removed from the default models in `Advanced Settings`.
- Ensured that Mixtral is the default model for new users. 
- Ensured new conversations will start with Mixtral. 

Example | Example | Example | Example
-|-|-|-
![Screenshot 2024-06-04 at 2 10 33 PM](https://github.com/brave/brave-browser/assets/17885425/ad9f1747-bd88-4a0b-8d7d-14b6e03f774d) | ![Screenshot 2024-06-04 at 2 24 00 PM](https://github.com/brave/brave-browser/assets/17885425/ca99ef18-bac5-4abb-bed1-910234a9b280) | ![Screenshot 2024-06-04 at 2 24 21 PM](https://github.com/brave/brave-browser/assets/17885425/9a6fcf5a-2cab-4489-a342-c956c2dc5050) | ![Screenshot 2024-06-04 at 2 24 34 PM](https://github.com/brave/brave-browser/assets/17885425/d032e3d7-38ac-4d32-be35-c47924854144)


**Case: Existing Users (Non-premium)**

- Ensured that selecting `Claude Instant` and upgrading will default to `Claude 3 Haiku` for premium users.
- Ensured that `Claude Instant` is removed from the language selection menu.
- Ensured that non-premium users will still hit the paywall when using Haiku model.
- Ensured that switching to the free model enables user to continuing using chat after hitting the rate limit.

Example | Example
-|-
![Screenshot 2024-06-04 at 2 35 20 PM](https://github.com/brave/brave-browser/assets/17885425/3e90679f-6e64-4b50-854c-3004af9a961c) | ![Screenshot 2024-06-04 at 2 35 27 PM](https://github.com/brave/brave-browser/assets/17885425/599fa812-109a-4b59-857b-632dce894931)


**Case: Existing Users (Premium)**

- Ensured that selecting `Claude Instant` and upgrading will default to `Claude 3 Sonnet` for premium users.
- Ensured that `Claude Instant` is removed from the language selection menu.
- Ensured that new conversations start with Sonnet.
- Ensured chatting after switching to another model without issues. 

Example | Example | Example
-|-|-
![Screenshot 2024-06-04 at 11 40 25 AM](https://github.com/brave/brave-browser/assets/17885425/0eed7169-cedb-4338-bdfe-8aeaf22530dd) | ![Screenshot 2024-06-04 at 11 48 33 AM](https://github.com/brave/brave-browser/assets/17885425/0f5d07e7-93e1-4979-bb2b-bbda2eb59c69) | ![Screenshot 2024-06-04 at 11 48 39 AM](https://github.com/brave/brave-browser/assets/17885425/74d7245f-fdf0-4353-a81a-37cdff5305e4)

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using

```
Brave | 1.67.107 Chromium: 126.0.6478.26 (Official Build) beta (x86_64)
-- | --
Revision | 2831369c7540d93e98f38d9b463f6762430a7982
OS | macOS Version 11.7.10 (Build 20G1427)
```
